.github/workflows/ttnn-run-sweeps.yaml

```diff
@@ -2,40 +2,127 @@ name: "ttnn - Run sweeps"
+      - name: Run ttnn sweeps generation (all sweeps)
+        if: ${{ github.event_name == 'schedule' || github.event.inputs.sweep_name == 'ALL SWEEPS' }}
+        run: |
+          python tests/sweep_framework/parameter_generator.py --elastic cloud --tag ci-main --explicit
+
+  ttnn-run-sweeps:
+    needs: ttnn-generate-sweeps
     strategy:
       # Do not fail-fast because we need to ensure all tests go to completion
       # so we try not to get hanging machines
       fail-fast: false
       matrix:
-        arch: [grayskull, wormhole_b0]
+        test-group:
+          [
+            {
+              name: "Grayskull E150 Sweeps",
+              arch: grayskull,
+              runs-on: ["cloud-virtual-machine", "E150", "in-service"],
+              tt-smi-cmd: "tt-smi-metal -r 0"
+            },
+            {
+              name: "Wormhole N150 Sweeps",
+              arch: wormhole_b0,
+              runs-on: ["cloud-virtual-machine", "N150", "in-service"],
+              tt-smi-cmd: "tt-smi-metal -r 0"
+            },
+            {
+              name: "Wormhole N300 Sweeps",
+              arch: wormhole_b0,
+              runs-on: ["cloud-virtual-machine", "N300", "in-service"],
+              tt-smi-cmd: "tt-smi-metal -r 0"
+            }
+          ]
     env:
       TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
-      ARCH_NAME: ${{ matrix.arch }}
+      ARCH_NAME: ${{ matrix.test-group.arch }}
       ELASTIC_USERNAME: ${{ secrets.SWEEPS_ELASTIC_USERNAME }}
       ELASTIC_PASSWORD: ${{ secrets.SWEEPS_ELASTIC_PASSWORD }}
+      TT_SMI_RESET_COMMAND: ${{ matrix.test-group.tt-smi-cmd }}
       LD_LIBRARY_PATH: ${{ github.workspace }}/build/lib
     environment: dev
     timeout-minutes: 720
-    runs-on: ${{ matrix.arch }}
+    runs-on: ${{ matrix.test-group.runs-on }}
     steps:
       - uses: tenstorrent-metal/metal-workflows/.github/actions/checkout-with-submodule-lfs@v2.0.0
       - name: Set up dynamic env vars for build
         run: |
           echo "TT_METAL_HOME=$(pwd)" >> $GITHUB_ENV
       - uses: ./.github/actions/prepare-metal-run
         with:
-          arch: ${{ matrix.arch }}
-      - name: Run ttnn sweeps
+          arch: ${{ matrix.test-group.arch }}
+      - name: Setup ttnn sweeps runner environment
         run: |
           source ${{ github.workspace }}/python_env/bin/activate
           cd $TT_METAL_HOME
           export PYTHONPATH=$TT_METAL_HOME
-          python tests/sweep_framework/runner.py --elastic cloud
+      - name: Run ttnn sweeps (single sweep)
+        if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.sweep_name != 'ALL SWEEPS' }}
+        run: |
+          python tests/sweep_framework/runner.py --module-name ${{ github.event.inputs.sweep_name }} --elastic cloud --tag ci-main
+      - name: Run ttnn sweeps (all sweeps)
+        if: ${{ github.event_name == 'schedule' == '' || github.event.inputs.sweep_name == 'ALL SWEEPS' }}
```

Comment:
This looks like a syntax error - is the code taht I gave you before? If it is, I clearly should be fired

Reply:
Nope, copy-paste error. Fixed 😄